queries.sql

```diff
@@ -113,3 +113,82 @@ join owners on owners.id = animals.owner_id
 group by owners.full_name 
 order by maximum 
 DESC LIMIT 1;
+
+-- Who was the last animal seen by William Tatcher?
+
+SELECT animals.name
+FROM visits 
+JOIN animals on animals.id = visits.animals_id
+JOIN vets on vets.id = visits.vets_id
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits_date 
+DESC LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+
+SELECT vets.name as vets_name, count(*)
+FROM visits
+JOIN animals on animals.id = visits.animals_id
+JOIN vets on vets.id = visits.vets_id 
+WHERE vets.name = 'Stephanie Mendez'
+GROUP BY vets.name
+
+-- List all vets and their specialties, including vets with no specialties.
+
+SELECT vets.name as vets_name, species.name as species_name
+FROM vets
+LEFT JOIN specializations on vets.id = specializations.vets_id
+LEFT JOIN species on species.id = specializations.species_id
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+
+SELECT animals.name
+FROM visits
+JOIN vets on vets.id = visits.vets_id
+JOIN animals on animals.id = visits.animals_id
+WHERE vets.name = 'Stephanie Mendez' 
+and visits_date between 'April-01-2020' and 'August-30-2020'
+
+-- What animal has the most visits to vets?
+
+SELECT animals.name, count(*) as num_of_times
+FROM visits
+JOIN vets on vets.id = visits.vets_id
+JOIN animals on animals.id = visits.animals_id
+group by animals.name
+order by num_of_times desc limit 1
+
+-- Who was Maisy Smith's first visit?
+
+SELECT animals.name
+FROM visits
+JOIN vets on vets.id = visits.vets_id
+JOIN animals on animals.id = visits.animals_id
+where vets.name = 'Maisy Smith'
+order by visits_date
+limit 1
```

Comment:
- [ ] The first animal to visit Maisy Smith is `Boarmon`, on the 24th of Jan, 2019. Your query returns the correct result but the screenshot you attached is wrong. Kindly update it. 👍

Reply:
I think there is a mixed up in my database when trying to insert the ids, that is  the reason it's displaying another animal name

Reply:
@Fayob, I understand. If you think there is a mix-up, kindly cross-check the data inserted into the tables and ensure you insert the correct data. The truth is if you insert wrong data into tables, you will automatically return wrong results for your queries.